driver/src/metrics/http_metrics.rs

```diff
@@ -0,0 +1,249 @@
+use anyhow::Result;
+use prometheus::{IntGauge, IntGaugeVec, Opts, Registry};
+use std::any::{Any, TypeId};
+use std::collections::HashMap;
+use std::fmt::{self, Debug, Formatter};
+use std::sync::Arc;
+use std::time::Duration;
+
+/// A registry for all HTTP related metrics.
+#[derive(Debug)]
+pub struct HttpMetrics {
+    labeled: HashMap<TypeId, SubsystemMetrics<IntGaugeVec>>,
+    unlabeled: HashMap<TypeId, SubsystemMetrics<IntGauge>>,
```

Comment:
Has to be histogram instead of gauge.

Reply:
From what I understand, histograms require us to know how to split it up into buckets, I'm not sure what sizes and latencies to use :(

Reply:
Gauge would not work in this case so it has to be Histogramm. With a Gauge you only have a single value so every time we did one http request we would overwrite the old value.

We do have to pick bucket sizes but I wouldn't overthink it. We can use this function https://docs.rs/prometheus/0.8.0/prometheus/fn.exponential_buckets.html to make buckets. `exponential_buckets(0.5, 2, 5)` would  return `[0.5, 1, 2, 4, 8]` second buckets which seems fine.